AUTH.md

```diff
@@ -0,0 +1,134 @@
+# Authentication and Authorization in WITSML Explorer
+
+## WITSML server credentials flow
+Every request run against a WITSML server is run from the backend and needs to be authenticated.
+Basic auth is required for a lot of servers, so that is currently the main way WE authenticate against them.
+
+Most actions done by the user in WE involves fetching or writing data to external WITSML servers. All these requests require credentials to be provided. 
+It would be a very bad user experience if the user would have to provide credentials for every request. 
+Therefore an encrypted version of the passwords is saved in memory on the frontend.
+
+The webapp only stores them for a limited amount of time, and will provide them for every request involving WITSML servers. 
+
+The backend has a [Data Protection](https://docs.microsoft.com/en-us/aspnet/core/security/data-protection/introduction) storage running in memory, which is used for creating the encrypted passwords as well as decrypting them (only possible for that running instance).
+Whenever a request is run towards a WITSML server, the backend will decrypt the password, and use it when running the request against the given WITSML server.
+
+This is how the flow is when a user has selected a server and will need to authenticate against it. After this is done, a fresh list of wells is fetched.  
+
+<img src="./credentials-flow.svg">
+
+## OAuth2
+OAuth2 authorization code flow and system credentials fetched from keyvault can also be used for a simplified end user experience. Examples outlined below for Azure AAD and Azure keyvault.
```

Comment:
"Examples _are_ outlined below"?
Isn't it just Azure AD, not AAD?

Reply:
Correct, will fix